### PR TITLE
Add handler helpers

### PR DESCRIPTION
**Pull request by ohyde**
For a small lambda I found this to be a fun way to switch between events.

E.g.

`return R.cond([
		[getAPI, getHandler(callback)],
		[postAPI, createHandler(callback)],
		[putAPI, updateHandler(callback)],
		[deleteAPI, deleteHandler(callback)],
		[R.T, defaultHandler(callback)]
	])(event);`

### CODE REVIEW

File: src/utils/api/handlerUtils.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const methodGET = R.compose(R.equals('GET'), R.propOr(null, "method"))
```
Comment (lucasschejtman) [inline src/utils/api/handlerUtils.js]: http://ramdajs.com/docs/#propEq

File: src/utils/api/handlerUtils.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+const methodPOST = R.compose(R.equals('POST'), R.propOr(null, "method"))
+const methodPUT = R.compose(R.equals('PUT'), R.propOr(null, "method"))
+const methodDELETE = R.compose(R.equals('DELETE'), R.propOr(null, "method"))
+
+const resourceBase = R.compose(R.equals('/'), R.propOr(null, "resource"))
+export const createResourceTest = (resource) => R.compose(R.equals(resource), R.propOr(null, "resource"))
+
+// expect an object with resource and method keys e.g. { resource: '/', method: 'GET'g}
+export const getAPI = R.allPass([methodGET, resourceBase]);
```
Comment (lucasschejtman) [inline src/utils/api/handlerUtils.js]: why does it needs to be base ('/') ? Also, you can define diff handlers for different verbs in lambda

Comment (ohyde) [reply src/utils/api/handlerUtils.js]: It doesn't, it was just as an example ...you might have 
`
const resourceInvoice = R.propEq('resource', 'invoice');`


and then 
`export const getInvoice = R.allPass([methodGET, resourceInvoice]);`

Comment (ohyde) [reply src/utils/api/handlerUtils.js]: Are you meaning this? https://serverless.com/framework/docs/providers/aws/guide/events/

Comment (ohyde) [reply src/utils/api/handlerUtils.js]: oh cool, I'll have a play :)

https://serverless.com/framework/docs/providers/aws/events/apigateway/

File: src/utils/api/handlerUtils.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
 // Built for AWS Lambda Function Event Object
 
-const methodGET = R.compose(R.equals('GET'), R.propOr(null, "method"))
-const methodPOST = R.compose(R.equals('POST'), R.propOr(null, "method"))
-const methodPUT = R.compose(R.equals('PUT'), R.propOr(null, "method"))
-const methodDELETE = R.compose(R.equals('DELETE'), R.propOr(null, "method"))
+const methodGET = R.propEq('method', 'GET');
```
Comment (lucasschejtman) [inline src/utils/api/handlerUtils.js]: you could have `const methodProp = R.propEq('method')`

File: src/utils/api/handlerUtils.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+// Built for AWS Lambda Function Event Object
+
+const methodGET = R.propEq('method', 'GET');
```
Comment (lucasschejtman) [inline src/utils/api/handlerUtils.js]: you could have `const methodProp = R.propEq('method')`

**Comment by lucasschejtman** [comment]
https://github.com/ohyde/ezM/blob/master/server/serverless-invoice/serverless.yml#L29

**Review by kahyde — APPROVED** [review]

**Pull request merged**